src/lib.rs

```diff
@@ -395,9 +397,45 @@ impl<M: Manager> Pool<M> {
                 }
                 _ => (),
             }
+            self.start_conn_cleaner(&mut internals);
         }
+        internals.config.max_lifetime = max_lifetime;
+    }
 
-        if max_lifetime.is_some() && internals.num_open > 0 && internals.cleaner_ch.is_none() {
+    /// Sets the maximum idle lifetime of connections in the pool.
+    ///
+    /// - `None` means reuse forever.
+    /// - Defaults to `None`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `max_idle_lifetime` is the zero `Duration`.
+    pub async fn set_conn_max_idle_lifetime(&self, max_idle_lifetime: Option<Duration>) {
+        assert_ne!(
+            max_idle_lifetime,
+            Some(Duration::from_secs(0)),
+            "max_idle_lifetime must be positive"
+        );
+
+        let mut internals = self.0.internals.lock().await;
+        if let Some(lifetime) = max_idle_lifetime {
+            match internals.config.max_idle_lifetime {
+                Some(prev) if lifetime < prev && internals.cleaner_ch.is_some() => {
+                    // FIXME
```

Comment:
Not sure what you had in mind for the result here @importcjj 
Returning a result would break the API for `set_conn_max_lifetime`